src/mono/mono/mini/aot-compiler.c

```diff
@@ -4371,10 +4331,14 @@ add_extra_method_full (MonoAotCompile *acfg, MonoMethod *method, gboolean prefer
 	}
 
 	if ((acfg->aot_opts.dedup || acfg->aot_opts.dedup_include) && mono_aot_can_dedup (method)) {
-		mono_dedup_cache_method (acfg, method);
-
-		if (!acfg->dedup_emit_mode)
+		if (acfg->aot_opts.dedup) {
```

Comment:
Parent `if` might not be needed if the nested `if`/`else if` statements are expanded with `&& mono_aot_can_dedup (method)`